modules/Movecraft/src/main/java/net/countercraft/movecraft/async/translation/TranslationTask.java

```diff
@@ -192,7 +193,8 @@ protected void execute() throws InterruptedException, ExecutionException {
                 int centreMinY = oldHitBox.getMinYAt(midPoint.getX(), midPoint.getZ());
                 int groundY = centreMinY;
                 World w = craft.getWorld();
-                while (w.getBlockAt(midPoint.getX(), groundY - 1, midPoint.getZ()).getType().isAir() || craft.getType().getMaterialSetProperty(CraftType.PASSTHROUGH_BLOCKS).contains(w.getBlockAt(midPoint.getX(), groundY - 1, midPoint.getZ()).getType())){
+                int minimumHeight = WorldUtils.getWorldMinHeightLimit(w);
+                while (groundY - 1 >= minimumHeight && w.getBlockAt(midPoint.getX(), groundY - 1, midPoint.getZ()).getType().isAir() || craft.getType().getMaterialSetProperty(CraftType.PASSTHROUGH_BLOCKS).contains(w.getBlockAt(midPoint.getX(), groundY - 1, midPoint.getZ()).getType())){
```

Comment:
While we don't have an explicit line length maximum, the condition in this while loop should be split across multiple lines to improve readability.  For example:
```java
while (groundY - 1 >= minimumHeight
        && w.getBlockAt(midPoint.getX(), groundY - 1, midPoint.getZ()).getType().isAir()
        || craft.getType().getMaterialSetProperty(CraftType.PASSTHROUGH_BLOCKS).contains(w.getBlockAt(midPoint.getX(), groundY - 1, midPoint.getZ()).getType())) {
    groundY--;
}
```

Reply:
This clarification shows a clear problem, your statement is of the following form `a && b || c`, which will evaluate the AND before the OR (`(a && b) || c`), rather than what I suspect the intention was to be `a && (b || c)`.  Either way the intention is, you should add clarifying parenthesis such that somebody without intimate knowledge of operator precedence in java can understand the condition.